src/ucp/core/ucp_worker.c

```diff
@@ -110,9 +110,13 @@ static ucs_mpool_ops_t ucp_rkey_mpool_ops = {
     kh_int64_hash_func((uintptr_t)(_uct_ep))
 
 
-KHASH_IMPL(ucp_worker_discard_uct_ep_hash, uct_ep_h, char, 0,
+KHASH_IMPL(ucp_worker_discard_uct_ep_hash, uct_ep_h, ucp_request_t*, 1,
            ucp_worker_discard_uct_ep_hash_key, kh_int64_hash_equal);
 
+/* Forward declarations to use it inside ucp_worker_discard_remove_filter() */
+static unsigned ucp_worker_discard_uct_ep_progress(void *arg);
+static unsigned ucp_worker_discard_uct_ep_destroy_progress(void *arg);
+
```

Comment:
can we move these functions ahead, instead?

Reply:
not sure that I understand it correctly